src/benchmark/benchmarkFn.ts

```diff
@@ -0,0 +1,168 @@
+import fs from "node:fs";
+import path from "node:path";
+import {getCurrentSuite, Suite, SuiteCollector} from "@vitest/runner";
+import {createChainable} from "@vitest/runner/utils";
+import {store} from "./globalState.js";
+import {BenchmarkOpts} from "../types.js";
+import {runBenchFn} from "./runBenchmarkFn.js";
+import {optionsDefault} from "../cli/options.js";
+
+export type BenchmarkRunOptsWithFn<T, T2> = BenchmarkOpts & {
+  id: string;
+  fn: (arg: T) => void | Promise<void>;
+  before?: () => T2 | Promise<T2>;
+  beforeEach?: (arg: T2, i: number) => T | Promise<T>;
+};
+
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+
+export function getRootSuite(suite: Suite | SuiteCollector): Suite {
+  if (suite.type === "collector") {
+    return getRootSuite(suite.tasks[0] as Suite);
+  }
+
+  if (suite.name === "") {
+    return suite;
+  } else if (suite.suite) {
+    getRootSuite(suite.suite);
+  } else {
+    return suite;
+  }
+
+  throw new Error("Can not find root suite");
+}
+
+export const bench = createBenchmarkFunction(function <T, T2>(
+  this: Record<"skip" | "only", boolean | undefined>,
+  idOrOpts: string | PartialBy<BenchmarkRunOptsWithFn<T, T2>, "fn">,
+  fn?: (arg: T) => void | Promise<void>
+) {
+  const {fn: benchTask, ...opts} = coerceToOptsObj(idOrOpts, fn);
+  const currentSuite = getCurrentSuite();
+
+  const globalOptions = store.getGlobalOptions() ?? {};
+  const parentOptions = store.getOptions(getCurrentSuite()) ?? {};
+  const options = {...globalOptions, ...parentOptions, ...opts};
+  const {timeoutBench, maxMs, minMs} = options;
+
+  let timeout = timeoutBench ?? optionsDefault.timeoutBench;
+  if (maxMs && maxMs > timeout) {
+    timeout = maxMs * 1.5;
```

Comment:
This logic is been copied as it is from earlier implementation. 

If our default timeout is lower than a specific benchmark `maxMs` then it will always timeout. So we had to increase the timeout for that test.